homeassistant/components/sonos/const.py

```diff
@@ -2,3 +2,11 @@
 
 DOMAIN = "sonos"
 DATA_SONOS = "sonos_media_player"
+
+TYPE_ARTIST = "artists"
```

Comment:
We should define these types in `media_player/const.py` and they should be generic for all media players items. That way we can better style it in the frontend because we know the type.

You should be able to derive if something is a libary etc from the ID. If the default IDs can't do that, you can always prefix the internal type to the ID `artists/{ORIG_ID}`

CC @jjlawren

Reply:
The frontend is currently intentionally agnostic on the data it receives. Maybe we should save having explicit "types" until there's a use-case?

Reply:
Yeah, we should. A season of a TV show is not the same as a music album.

Reply:
This has been worked on in https://github.com/home-assistant/core/pull/39240 and needs to be properly mapped for Sonos.

Reply:
Spotify has been merged, we should update this PR with the changes.

Reply:
Yes, already in the works.

Reply:
Shouldn't these be removed now ?

Reply:
Well, I haven't pushed anything meaningful yet. Still working on it. I found a few flaws.

Reply:
This is now fully functional as far as I've tested.